src/dissect/dissect.c

```diff
@@ -896,21 +899,32 @@ static int action_dissect(DissectedImage *m, LoopDevice *d) {
                 printf("            %s initrd\n",
                        COLOR_MARK_BOOL(!strv_isempty(m->initrd_release)));
 
-                r = get_sysext_scopes(m, &sysext_scopes);
+                r = (m->image_class == IMAGE_SYSEXT) ? get_extension_scopes(m, &sysext_scopes) : get_extension_scopes(m, &confext_scopes);
                 if (r < 0)
-                        return log_error_errno(r, "Failed to parse SYSEXT_SCOPE: %m");
-
-                printf("            %s extension for system\n",
-                       COLOR_MARK_BOOL(strv_contains(sysext_scopes, "system")));
-                printf("            %s extension for initrd\n",
-                       COLOR_MARK_BOOL(strv_contains(sysext_scopes, "initrd")));
-                printf("            %s extension for portable service\n",
-                       COLOR_MARK_BOOL(strv_contains(sysext_scopes, "portable")));
+                        return log_error_errno(r, "Failed to parse scope: %m");
+
+                if (m->image_class == IMAGE_CONFEXT) {
+                        printf("            %s confext extension for system\n",
```

Comment:
the only difference is 'confext' or 'sysext' so make that conditional instead of duplicating the block